### PR TITLE
Support for Windows, MacOS with psutil; color modes; refactoring; repeating mode

### PR DESCRIPTION
**Pull request by mzzfederico**
- fixed the setup.sh so it works without issues in the Bash for Windows
- added in 3 modes: greyscale (from dark to white), colorscale (applies a color by frequency, dark to light), color (all words same color)
- refactored the code a bit into at least a number of functions
- fixed margins for Windows and Mac OS (at least I think)
- added in "repeating" option in the JSON, so you can fill the wallpaper by repeating the words (useful on Windows since it has very few words)

color
mono
grey

### CODE REVIEW

File: generateWallpaper.py [hunk 1]
```diff
@@ -1,76 +1,131 @@
...
+#!/usr/bin/env python
+import re
+import operator
+import psutil
+import json
+import os
+from sys import platform
+from PIL import Image
+from wordcloud import WordCloud
+
```
Comment (dsluo) [inline generateWallpaper.py:10]: You forgot a `subprocess` import.

File: generateWallpaper.py [hunk 2]
```diff
@@ -1,76 +1,131 @@
...
+    if platform == "linux" or platform == "linux2":
+        try:
+            width, height = (
+                (os.popen("xrandr | grep '*'").read()).split()[0]).split("x")
+        except:
+            pass
+    elif platform == "darwin":
+        try:
+            results = str(subprocess.Popen(
+                ['system_profiler SPDisplaysDataType'], stdout=subprocess.PIPE, shell=True).communicate()[0])
```
Comment (dsluo) [inline generateWallpaper.py:24]: Consider just using [`subprocess.run`](https://github.com/anirudhajith/process-wallpaper/pull/15/files#r317433571).

**Comment by mzzfederico** [comment]
@dsluo thanks for the suggestions, I'm pretty new to Python so I'll have to look into that! Since tomorrow I'm back to work I might not have time to do much more :(

Edit: and I mistakenly pressed close, oops

**Comment by dsluo** [comment]
I would prefer that pull requests be separated into one request per big change, i.e. the psutil usage/color modes/repeating modes would each be their own separate pull requests. But it's probably fine for now.

File: generateWallpaper.py [hunk 3]
```diff
@@ -1,76 +1,131 @@
...
+screen_sizes = init_screen_sizes(configJSON)
+dump_screen_sizes(configJSON, screen_sizes)
+
+resourceDict = get_resources()
+
+width = screen_sizes[0]
+height = screen_sizes[1]
+cloud = WordCloud(
+    background_color=background,
+    width=int(width - (2 * margin)),
```
Comment (dsluo) [inline generateWallpaper.py:122]: Just running this gives me a `TypeError`
```
$ python generateWallpaper.py
```
```
Traceback (most recent call last):
  File "generateWallpaper.py", line 122, in <module>
    width=int(width - (2 * margin)),
TypeError: unsupported operand type(s) for -: 'str' and 'int'
```

**Pull request closed without merging**